File: benchmarks/oopsla25/subjects/292.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f292(const char* s) {
    if (strlen(s) == 0) {
        return;
    }

    char* bi = strchr(s, 'b');
    if (bi == s) {
        assert(strlen(s) == 1);
    } else if (bi == s + 1) {
        assert(s[0] == 'a');
        assert(strlen(s) == 2);
    } else {
        assert(strcmp(s, "a") == 0);
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f292(my_string);
  return 0;
}

File: benchmarks/oopsla25/subjects/292.py
import sys

def f292(s: str):
  if len(s) == 0:
    return
  bi = s.find("b")
  if bi == 0:
    assert len(s) == 1
  elif bi == 1:
    assert s.index("a") == 0
    assert len(s) == 2
  else:
    assert s == "a"

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f292(my_string)
